solutions/usaco-570.mdx

```diff
@@ -2,72 +2,374 @@
 id: usaco-570
 source: USACO Silver 2015 December
 title: Switching on the Lights
-author: Óscar Garries
+author: Akshar Yeccherla
 ---
 
-[Official Analysis](http://www.usaco.org/current/data/sol_lightson_silver_dec15.html)
+[Official Analysis (Java)](http://www.usaco.org/current/data/sol_lightson_silver_dec15.html)
 
-<LanguageSection>
-<CPPSection>
+## Explanation
+
+We can traverse through the rooms that are lit and connected to the room $(1,1)$ using floodfill, since we are essentially traversing a connected component. We can then turn on all the switches in each room we visit.
+
+However, we run into an issue when a newly lit room that becomes part of the main connected component can't be visited since the floodfill already visited the lit rooms neighboring it. 
 
+To account for this, we start a floodfill from newly lit rooms if they are connected to the connected component that contains $(1,1)$ (i.e. it's accessible to us). By keeping track of the number of rooms we light up along the way, we get our answer. 
 ## Implementation
 
+**Time Complexity:** $\mathcal{O}(N^2 + M)$
+
+<LanguageSection>
+<CPPSection>
+
 ```cpp
 #include <bits/stdc++.h>
-
 using namespace std;
 
-const int MX = 1e2;
+int N;
+int litRooms = 1;
+bool visited[101][101];
+bool illuminated[101][101];
+vector<pair<int,int>> switches[101][101];
```

Comment:
1. put in constant
2. make 0-indexed if possible
3. vectors
(2 & 3 are optional)